app/vmalert/rule/recording.go

```diff
@@ -91,6 +92,7 @@ func NewRecordingRule(qb datasource.QuerierBuilder, group *Group, cfg config.Rul
 		GroupID:   group.GetID(),
 		GroupName: group.Name,
 		File:      group.File,
+		Debug:     cfg.Debug,
```

Comment:
cfg.Debug should be propagated to the query builder on L:101, similarly to alerting rule